File: diagonalDifference.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

int main() {
  int n;
  scanf("%d", &n);
  int a[n][n];
  int diagonalLeftSum = 0;
  int digonalRightSum = 0;

  for(int a_i = 0; a_i < n; a_i++) {
    for(int a_j = 0; a_j < n; a_j++) {
      scanf("%d", &a[a_i][a_j]);
    }
  }
  for(int j = 0; j < n; j++) {
    for(int k = 0; k < n; k++) {
      if (j == k) {
	diagonalLiftSum += a[j][k];
      }
      if (j + k == n - 1) {
	diagonalRightSum += a[j][k];
      }
    }
  }
  int difference = abs(diagonalRightSum - diagonaleftSum);
  printf("%i", difference);
  return 0;
}

File: diagonaldifference.py
#!/bin/python3

import sys


n = int(input().strip())
a = []
for a_i in range(n):
    a_t = [int(a_temp) for a_temp in input().strip().split(' ')]
    a.append(a_t) 
diagonalLeftSum = 0
diagonalRightSum = 0

for j in range(n):
    for k in range(n):
        if j == k:
            diagonalLeftSum += a[j][k]
        if j + k == n - 1:
            diagonalRightSum += a[j][k]
difference = abs(diagonalLeftSum - diagonalRightSum)
print(difference)
